cinema/views.py

```diff
@@ -1 +1,60 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Movie, Actor, Genre, CinemaHall, MovieSession
+from cinema.serializers import (
+    MovieSerializer,
+    MovieListSerializer,
+    MovieDetailSerializer,
+    ActorSerializer,
+    GenreSerializer,
+    CinemaHallSerializer,
+    MovieSessionDetailSerializer,
+    MovieSessionListSerializer,
+    MovieSessionCreateSerializer,
+)
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all().prefetch_related("actors", "genres")
+    serializer_class = MovieSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieDetailSerializer
+
+        return MovieSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = (
+        MovieSession.objects.all()
+        .select_related("cinema_hall", "movie")
+        .prefetch_related("movie__actors", "movie__genres")
+    )
+    serializer_class = MovieSessionDetailSerializer
```

Comment:
There is no need to define `serializer_class` attribute in case you implement `get_serializer_class` method